commands/create.go

```diff
@@ -458,3 +480,101 @@ func tlsPath(c CommandLine, flag string, defaultName string) string {
 
 	return filepath.Join(mcndirs.GetMachineCertDir(), defaultName)
 }
+
+func updateUserdataFile(driverOpts *rpcdriver.RPCFlags, userdataFlag, customInstallScript string) error {
+	var userdataContent []byte
+	var err error
+	userdataFile := driverOpts.String(userdataFlag)
+
+	if userdataFile == "" {
```

Comment:
This is hit when setting flag with no value ? I don't get this case.

Reply:
There are two cases we are trying to handle if the driver supports user data.
1. The user has passed a user data file to the driver. Then we want to add the "install script" to the user-provided user data. That is NOT this case.
2. The user has not provided a user data file. Then we just set our "install script" file as the user data file. That IS this case.

Reply:
That clarifies part of it, now my my confusion is around the two variables. If the user has _not_ provided a userdataFile why are we checking the userdataFlag for a specific string below, when userdataFile comes from userdataFlag ? There is some context around this flag/process I am missing.

Reply:
Oh, I'm sorry I forgot that case.
2B. The user has not provided a user data file, but the driver expects the user data file to be in `cloud-config` format. The custom install script is provided a script, so by setting `userdataContent` on line 501 we coerce the custom install script into `cloud-config` format.

Reply:
Oh I think I understand now, really my confusion was that userdata flags can be non-nil when userdata files are nil.